.github/workflows/test.yml

```diff
@@ -102,3 +99,58 @@ jobs:
             ${{ steps.setup-haskell.outputs.cabal-store }}
             dist-newstyle
           key: ${{ steps.cache.outputs.cache-primary-key }}
+
+  doc:
+    name: Publish haddocks to GitHub Pages
+    needs: test
+    env:
+      ghc: "9.10.1"
+      os: "ubuntu-24.04"
+    # For some reason, can't use ${{ env.os }} here, so just be sure to keep
+    # this in sync with the line above.
+    runs-on: "ubuntu-24.04"
+    concurrency:
+      # Only run one of these at a time because they update the global pages.
+      group: haddock
+      cancel-in-progress: false
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: haskell-actions/setup@v2
+        id: setup-haskell
+        with:
+          ghc-version: ${{ env.ghc }}
+      - uses: actions/cache/restore@v4
+        name: Restore Cabal cache
+        env:
+          key: cabal-${{ env.os }}-ghc${{ env.ghc }}
```

Comment:
This job should probably use a different cache key value than the main build job since the cached objects will be different.

Reply:
Ah, this is actually intentional, and for just the reverse reason! I think the cached objects (the dependencies) will be the same! Also note that this step doesn't *modify* the cache, it just reuses the pre-built cache from the `test` job. I can add a comment that makes this all more explicit.

Reply:
As for whether reusing the cache from `test` is sufficient for speedy doc builds, note that [this one took 18s](https://github.com/GaloisInc/parameterized-utils/actions/runs/13845141506/job/38741914356?pr=183)

Reply:
OK, I missed the fact that you have no cache/save action below, so this will always rebuild the current project, but can take advantage of the cached dependencies.  Please document this, either here or in dev documentation somewhere.

Reply:
Is this comment sufficient, or is there more I could add? https://github.com/GaloisInc/parameterized-utils/pull/183/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R131